dev/ProgressBar/ProgressBar.cpp

```diff
@@ -0,0 +1,171 @@
+
+    m_layoutRoot.set(GetTemplateChildT<winrt::Grid>(s_LayoutRootName, controlProtected));
+    m_progressBarIndicator.set(GetTemplateChildT<winrt::Rectangle>(s_ProgressBarIndicatorName, controlProtected));
+
+    UpdateStates();
+}
+
+void ProgressBar::OnSizeChanged(const winrt::IInspectable&, const winrt::IInspectable&)
+{
+    SetProgressBarIndicatorWidth();
+    if (m_shouldUpdateWidthBasedTemplateSettings)
+    {
+        UpdateWidthBasedTemplateSettings();
+    }
+}
+
+void ProgressBar::OnRangeBasePropertyChanged(const winrt::DependencyObject& sender, const winrt::DependencyProperty& args)
+{
+    // NOTE: This hits when the Value property changes, because we called RegisterPropertyChangedCallback.
+
+    SetProgressBarIndicatorWidth();
+}
+
+void ProgressBar::OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    // NOTE: This hits when IsIndeterminate changes because we set MUX_PROPERTY_CHANGED_CALLBACK to true in the idl.
+
+    UpdateStates();
+}
+
+void ProgressBar::OnShowPausedPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressBar::OnShowErrorPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressBar::UpdateStates()
+{
+    m_shouldUpdateWidthBasedTemplateSettings = false;
+    if (ShowError())
+    {
+        winrt::VisualStateManager::GoToState(*this, s_ErrorStateName, true);
+    }
+    else if (ShowPaused() && IsIndeterminate())
+    {
+        winrt::VisualStateManager::GoToState(*this, s_ErrorStateName, true); // Paused-Indeterminate state same visual treatment as Error state
+    }
+    else if (ShowPaused())
+    {
+        winrt::VisualStateManager::GoToState(*this, s_PausedStateName, true);
+    }
+    else if (IsIndeterminate())
+    {
+        m_shouldUpdateWidthBasedTemplateSettings = true;
+        UpdateWidthBasedTemplateSettings();
+        winrt::VisualStateManager::GoToState(*this, s_IndeterminateStateName, true);
+    }
+    else if (!IsIndeterminate())
+    {
+        SetProgressBarIndicatorWidth();
+        winrt::VisualStateManager::GoToState(*this, s_DeterminateStateName, true);
+    }
+}
+
+void ProgressBar::SetProgressBarIndicatorWidth()
+{
+    if (auto&& progressBar = m_layoutRoot.get())
+    {
+        if (auto&& progressBarIndicator = m_progressBarIndicator.get())
+        {
+            const double progressBarWidth = progressBar.ActualWidth();
+            const double maximum = Maximum();
+            const double minimum = Minimum();
+            const auto padding = Padding();
+
+            if (std::abs(maximum - minimum) > DBL_EPSILON)
+            {
+                const double maxIndicatorWidth = progressBarWidth - (padding.Left + padding.Right);
+
+                // checks if padding causes maxIndicatorWidth to be larger than progressBarWidth
+                if (progressBarWidth - padding.Left < maxIndicatorWidth)
```

Comment:
>if (progressBarWidth - padding.Left < maxIndicatorWidth) [](start = 16, length = 56)

So, this is true only if padding.right is negative, in which case the only change you are making is to ignore the padding.right value.  I'm not sure what this accomplishes